mod/members/start.php

```diff
@@ -15,181 +11,66 @@
 function members_init() {
 	elgg_register_page_handler('members', 'members_page_handler');
 
-	$item = new ElggMenuItem('members', elgg_echo('members'), 'members');
-	elgg_register_menu_item('site', $item);
-
-	$list_types = ['newest', 'alpha', 'popular', 'online'];
-
-	foreach ($list_types as $type) {
-		elgg_register_plugin_hook_handler('members:list', $type, "members_list_$type");
-		elgg_register_plugin_hook_handler('members:config', 'tabs', "members_nav_$type");
-	}
-}
-
-/**
- * Returns content for the "popular" page
- *
- * @param string      $hook        "members:list"
- * @param string      $type        "popular"
- * @param string|null $returnvalue list content (null if not set)
- * @param array       $params      array with key "options"
- * @return string
- */
-function members_list_popular($hook, $type, $returnvalue, $params) {
-	if ($returnvalue !== null) {
-		return;
-	}
-
-	$options = $params['options'];
-	$options['relationship'] = 'friend';
-	$options['inverse_relationship'] = false;
-	return elgg_list_entities_from_relationship_count($options);
-}
+	elgg_register_menu_item('site', [
+		'name' => 'members',
+		'text' => elgg_echo('members'),
+		'href' => 'members',
+	]);
 
-/**
- * Returns content for the "newest" page
- *
- * @param string      $hook        "members:list"
- * @param string      $type        "newest"
- * @param string|null $returnvalue list content (null if not set)
- * @param array       $params      array with key "options"
- * @return string
- */
-function members_list_newest($hook, $type, $returnvalue, $params) {
-	if ($returnvalue !== null) {
-		return;
-	}
-	return elgg_list_entities($params['options']);
+	elgg_register_plugin_hook_handler('register', 'menu:filter:members', 'members_register_filter_menu');
 }
 
 /**
- * Returns content for the "online" page
+ * Registers members filter menu items
  *
- * @param string      $hook        "members:list"
- * @param string      $type        "online"
- * @param string|null $returnvalue list content (null if not set)
- * @param array       $params      array with key "options"
- * @return string
- */
-function members_list_online($hook, $type, $returnvalue, $params) {
-	if ($returnvalue !== null) {
-		return;
-	}
-	return get_online_users();
-}
-
-/**
- * Returns content for the "alphabetical" page
+ * @param \Elgg\Hook $hook 'register', 'menu:filter'
```

Comment:
'menu:filter:members' NOT 'menu:filter'

Reply:
Use `@elgg_plugin_hook register menu:filter:members`

Reply:
i see you using that... but the usage of `@elgg_plugin_hook` is nowhere described... i do not even know its effect...

Reply:
changed it nonetheless

Reply:
@mrclay started using it, but it makes sense, we can then detect them with annotation parsers.